website/addons/s3/static/tests/s3NodeConfig.test.js

```diff
@@ -0,0 +1,435 @@
+        postEndpoint.response = postEndpoint.response.result;
+        var bucket = faker.internet.domainWord();
+        postEndpoint.response.bucket = bucket;
+        postEndpoint.response.has_bucket = true;
+        var endpoints = [
+            postEndpoint,
+            makeSettingsEndpoint()
+        ];
+        var server;
+        before(() => {
+            server = utils.createServer(sinon, endpoints);
+        });
+        after(() => {
+            server.restore();
+        });
+        it('submits the selected bucket to the server, and updates data on success', (done) => {
+            var vm = new s3NodeConfigVM('/api/v1/12345/s3/settings/', '', '/12345');
+            vm.fetchFromServer()
+                .always(function() {
+                    vm.selectedBucket(bucket);
+                    var promise = vm.selectBucket();
+                    promise.always(function() {
+                        assert.equal(vm.currentBucket(), bucket);
+                        done();
+                    });
+                });
+        });
+    });
+    describe('Authorization/Authentication: ', () => {
+        var deleteEndpoint = makeSettingsEndpoint({
+            user_has_auth: true,
+            user_is_owner: true,
+            node_has_auth: false
+        });
+        deleteEndpoint.method = 'DELETE';
+        deleteEndpoint.response = deleteEndpoint.response.result;
+        var importEndpoint = makeSettingsEndpoint({
+            node_has_auth: true,
+            user_has_auth: true,
+            user_is_owner: true
+        });
+        importEndpoint.method = 'POST';
+        importEndpoint.url = URLS.import_auth;
+        importEndpoint.response = importEndpoint.response.result;
+        var createEndpoint = makeSettingsEndpoint({
+            node_has_auth: true,
+            user_has_auth: true,
+            user_is_owner: true
+        });
+        createEndpoint.method = 'POST';
+        createEndpoint.url = URLS.create_auth;
+        createEndpoint.response = createEndpoint.response.result;
+        var endpoints = [
+            makeSettingsEndpoint({
+                user_has_auth: true,
+                user_is_owner: true,
+                node_has_auth: true
+            }),
+            deleteEndpoint,
+            importEndpoint,
+            createEndpoint
+        ];
+        var server;
+        beforeEach(() => {
+            server = utils.createServer(sinon, endpoints);
+        });
+        afterEach(() => {
+            server.restore();
+        });
+
+        describe('#_deauthorizeNodeConfirm', () => {
+            it('makes a delete request to the server and updates settings on success', (done) => {
+                var expected = endpoints[1].response;
+                var vm = new s3NodeConfigVM('/api/v1/12345/s3/settings/', '', '/12345');
+                vm.fetchFromServer()
+                    .always(function() {
+                        var promise = vm._deauthorizeNodeConfirm();
```

Comment:
Same here: can we test `_deauthorizeNodeConfirm()` without calling fetchFromServer?